sw/device/lib/dif/dif_hmac.h

```diff
@@ -0,0 +1,211 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 1.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_DIF_HMAC_DEVICE_H_
+#define SW_DEVICE_LIB_DIF_HMAC_DEVICE_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * Supported HMAC modes of operation.
+ */
+typedef enum dif_hmac_mode {
+  /** The HMAC mode */
+  kDifHmacModeHmac = 0,
```

Comment:
Matter of habit mostly. For the error enums it could simplify certain types of error checking, but conversations about leveraging the falsiness of `kOk = 0` for error checking seem to end in the consensus that it's a bad idea. I don't think there is any real reason to keep these enum-constants.